### PR TITLE
Add Prettier formatter for JS files

### PR DESCRIPTION
**Pull request by montogeek**
Continues https://github.com/webpack/webpack.js.org/pull/2247/

### CODE REVIEW

**Comment by anshumanv** [comment]
lmk if that needs to be closed 😄

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "printWidth": 100
+}
```
Comment (evilebottnawi) [inline .prettierrc]: I recommend use this too:
```
  trailingComma: 'es5',
  arrowParens: 'always',
```
- Less changes in review
- Use parens for consistently output
- Also i recommend don't use printWidth: `100` with spaces, it is can generate unreadable code

**Comment by montogeek** [comment]
@evilebottnawi @anshumanv @EugeneHlushko Ping

@anshumanv I would like to include all your changes, let's do it after merging this one.

**Comment by anshumanv** [comment]
WIll update 👍

**Comment by jeffin143** [comment]
What is the blockage here ?

**Comment by evilebottnawi** [comment]
@montogeek Why prettier is blocked? 

@chenxsan I think we need focus on this after webpack@5 upgrading

**Comment by chenxsan** [comment]
Sure, I can do it after the webpack v5 upgrade if @montogeek doesn't have time to continue it.

**Comment by montogeek** [comment]
There is no blocker at this moment, I will restart it instead, can do it soon :)

**Comment by chenxsan** [comment]
@montogeek Hi, do you still have time doing this? Otherwise I can take care of it.

**Comment by chenxsan** [comment]
Closed via https://github.com/webpack/webpack.js.org/pull/4518

**Pull request closed without merging**